### PR TITLE
Extend end-to-end tests

### PR DESCRIPTION
**Pull request by deniszh**
## What issue is this change attempting to solve?
Extending end to end tests beyond simple /render call - now we're testing fid and expand too, with different parameters and output formats.

It's a first and quick approach to e2e tests, we probably would need to follow #448 to make proper system though.

## How does this change solve the problem? Why is this the best approach?
I reworked e2e tests in hurl format. [Hurl](https://hurl.dev) is cli tool for doing http requests, which can use simple test format to write http tests too.  Why hurl is better then e.g. curl for that task?  it's described in [FAQ](https://hurl.dev/docs/frequently-asked-questions.html) and I pretty much agreed with motivation.

Please also take note that hurl tests are completely optional, it assumes that hurl is installed and you need to run `make test-hurl` to run them.

### CODE REVIEW

**Comment by deniszh** [comment]
Example of test output
```bash
± make test-e2e
./tests/system_test.sh
[+] Building 15.5s (19/19) FINISHED
 => [carbonapi-carbonapi internal] load build definition from Dockerfile                                                                  0.0s
 => => transferring dockerfile: 32B                                                                                                       0.0s
 => [carbonapi-go-carbon internal] load build definition from Dockerfile                                                                  0.0s
 => => transferring dockerfile: 32B                                                                                                       0.0s
 => [carbonapi-carbonapi internal] load .dockerignore                                                                                     0.0s
 => => transferring context: 2B                                                                                                           0.0s
 => [carbonapi-go-carbon internal] load .dockerignore                                                                                     0.0s
 => => transferring context: 2B                                                                                                           0.0s
 => [carbonapi-carbonapi internal] load metadata for docker.io/library/golang:1.19                                                        1.4s
 => [carbonapi-go-carbon internal] load build context                                                                                     0.1s
 => => transferring context: 233B                                                                                                         0.1s
 => [carbonapi-carbonapi 1/8] FROM docker.io/library/golang:1.19@sha256:660f138b4477001d65324a51fa158c1b868651b44e43f0953bf062e9f38b72f3  0.0s
 => [carbonapi-carbonapi internal] load build context                                                                                     0.1s
 => => transferring context: 181.68kB                                                                                                     0.1s
 => CACHED [carbonapi-go-carbon 2/4] RUN mkdir -p /data/graphite/whisper/                                                                 0.0s
 => CACHED [carbonapi-go-carbon 3/4] COPY ./docker/go-carbon/*.conf /etc/                                                                 0.0s
 => CACHED [carbonapi-go-carbon 4/4] RUN go install github.com/go-graphite/go-carbon@master                                               0.0s
 => [carbonapi-carbonapi] exporting to image                                                                                              0.5s
 => => exporting layers                                                                                                                   0.5s
 => => writing image sha256:60dcc721e7b69e5409e76fb14768541cf8b454f8afdd130c6dab0e0f103af60c                                              0.0s
 => => naming to docker.io/library/carbonapi-go-carbon                                                                                    0.0s
 => => writing image sha256:6018a56f88455eaaadd2e72cf0c19d1e484c06202782f1b1e006d6f05b98f4b5                                              0.0s
 => => naming to docker.io/library/carbonapi-carbonapi                                                                                    0.0s
 => CACHED [carbonapi-carbonapi 2/8] RUN apt-get update                                                                                   0.0s
 => CACHED [carbonapi-carbonapi 3/8] RUN apt-get install -y libcairo2-dev                                                                 0.0s
 => CACHED [carbonapi-carbonapi 4/8] WORKDIR /go/src/github.com/bookingcom/carbonapi                                                      0.0s
 => [carbonapi-carbonapi 5/8] COPY . .                                                                                                    0.7s
 => [carbonapi-carbonapi 6/8] COPY ./docker/carbonapi/carbonapi.yaml /etc/carbonapi.yaml                                                  0.0s
 => [carbonapi-carbonapi 7/8] COPY ./docker/carbonapi/carbonzipper.yaml /etc/carbonzipper.yaml                                            0.0s
 => [carbonapi-carbonapi 8/8] RUN make build                                                                                             12.6s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
Network carbonapi_default  Creating
Network carbonapi_default  Created
Container carbonapi-go-carbon-1  Creating
Container carbonapi-go-carbon-1  Created
Container carbonapi-carbonapi-1  Creating
Container carbonapi-carbonapi-1  Created
Container carbonapi-go-carbon-1  Starting
Container carbonapi-go-carbon-1  Started
Container carbonapi-carbonapi-1  Starting
Container carbonapi-carbonapi-1  Started
/opt/homebrew/bin/hurl
tests/hurl/00-render.hurl: Running [1/11]
tests/hurl/00-render.hurl: Success (1 request(s) in 21 ms)
tests/hurl/01-render.hurl: Running [2/11]
tests/hurl/01-render.hurl: Success (1 request(s) in 6 ms)
tests/hurl/10-find.hurl: Running [3/11]
tests/hurl/10-find.hurl: Success (1 request(s) in 5 ms)
tests/hurl/11-find.hurl: Running [4/11]
tests/hurl/11-find.hurl: Success (1 request(s) in 3 ms)
tests/hurl/12-find.hurl: Running [5/11]
tests/hurl/12-find.hurl: Success (1 request(s) in 3 ms)
tests/hurl/13-find.hurl: Running [6/11]
tests/hurl/13-find.hurl: Success (1 request(s) in 4 ms)
tests/hurl/14-find.hurl: Running [7/11]
tests/hurl/14-find.hurl: Success (1 request(s) in 3 ms)
tests/hurl/15-find.hurl: Running [8/11]
tests/hurl/15-find.hurl: Success (1 request(s) in 3 ms)
tests/hurl/20-expand.hurl: Running [9/11]
tests/hurl/20-expand.hurl: Success (1 request(s) in 2 ms)
tests/hurl/21-expand.hurl: Running [10/11]
tests/hurl/21-expand.hurl: Success (1 request(s) in 3 ms)
tests/hurl/22-expand.hurl: Running [11/11]
tests/hurl/22-expand.hurl: Success (1 request(s) in 1 ms)
--------------------------------------------------------------------------------
Executed files:  11
Succeeded files: 11 (100.0%)
Failed files:    0 (0.0%)
Duration:        67 ms

0
[+] Running 3/3
 ⠿ Container carbonapi-carbonapi-1  Removed                                                                                               0.1s
 ⠿ Container carbonapi-go-carbon-1  Removed                                                                                               0.1s
 ⠿ Network carbonapi_default        Removed                                                                                               0.0s
```

**Review by auguzun — DISMISSED** [review]

File: tests/hurl/20-expand.hurl [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# tesing expand
```
[the remark this answers is not in the document]
Comment (deniszh) [reply tests/hurl/20-expand.hurl]: fixed

File: tests/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# e2e test
+
+This directory contails carbonapi test files for e2e test for the go-carbon -> zipper -> api stack.
+
+Brings the system up, brings data, doing some render, find abd expand calls.
+Checks if data got in and out.
+
+Needs to be strated from Makefile in the root directory of the project.
```
[the remark this answers is not in the document]
Comment (deniszh) [reply tests/README.md]: fixed

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# e2e test
+
+This directory contails carbonapi test files for e2e test for the go-carbon -> zipper -> api stack.
+
+Brings the system up, brings data, doing some render, find abd expand calls.
```
[the remark this answers is not in the document]
Comment (deniszh) [reply tests/README.md]: fixed

File: tests/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# e2e test
+
+This directory contails carbonapi test files for e2e test for the go-carbon -> zipper -> api stack.
```
[the remark this answers is not in the document]
Comment (deniszh) [reply tests/README.md]: fixed

**Review by auguzun — APPROVED** [review]

**Review by grzkv — APPROVED** [review]

**Pull request merged**